### PR TITLE
LPS-140094 Update SearchContainer to filter the object entries already related

### PR DESCRIPTION
**Pull request by marcelabc**
Related issue: https://issues.liferay.com/browse/LPS-140094

Hi reviewer, 
I did not split the implementation because I thought that would be more confusing and cause compile errors. I will detail some points here. Let me know if there is any doubt, please.

To filter the related objects, it was necessary to:
- Get the related entries [_objectEntryLocalService.getOneToManyRelatedObjectEntries(...)]
- Create the filter string to compare the ID to not be equals to the ones already related (_getFilterString())
- Create a filter to exclude these entries using objectEntryResource.toFilter(...)
- Update Search with the new Filter

### CODE REVIEW

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:relevant

**Review by gabrielwas — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/item/selector/ObjectEntryItemSelectorView.java [hunk 1]
```diff
@@ -313,56 +339,101 @@ public ItemSelectorReturnType getItemSelectorReturnType() {
...
+				(ThemeDisplay)_httpServletRequest.getAttribute(
+					WebKeys.THEME_DISPLAY);
+
+			return new DefaultDTOConverterContext(
+				false, null, null, _httpServletRequest, objectEntryId,
+				_httpServletRequest.getLocale(), _contextUriInfo,
+				themeDisplay.getUser());
+		}
+
+		private String _getFilterString(List<ObjectEntry> objectEntryList) {
```
[the remark this answers is not in the document]
Comment (marcelabc) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/item/selector/ObjectEntryItemSelectorView.java:398]: @gabrielwas This method is used to create the parameter called "FilterString" in _objectEntryResource.toFilter(...) and this is the only use in this class. That's the reason of the name.
Do you think that would be better to rename?

**Comment by marcelabc** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/item/selector/ObjectEntryItemSelectorView.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (gabrielwas) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/item/selector/ObjectEntryItemSelectorView.java:398]: @marcelabc sorry about not being clear. I meant renaming the `objectEntryList` to `objectEntries`

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
I'm changing the label because the review result was only 1 rename. 
cc @gabrielwas (let me know if you didn't like this workflow)

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
Just started reviewing :)

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

**Pull request closed without merging**